src/main/java/org/worshipsongs/importer/TopicDao.java

```
@@ -8,55 +8,60 @@
-            if(resultSet.next()) {
+            ResultSet resultSet = statement.executeQuery("SELECT * FROM TOPICS where name = '" + topic.getTopic() + "';");
+            if (resultSet.next()) {
                 id = resultSet.getInt("id");
             }
             resultSet.close();
             statement.close();
+        } catch (Exception e) {
+            System.out.println("Exception:" + e);
         }
-        catch (Exception e) {
-            System.out.println("Exception:"+e);
-        }
-        return id;
+        topic.setId(id);
+        return topic;
     }
 
-    public boolean insertTopicSongs(Connection connection, Topic topic, int songId)
+    public boolean insertTopicSongs(Song song)
     {
         try {
             String query = "insert into songs_topics (song_id, topic_id) values (?, ?)";
             PreparedStatement preparedStatement = connection.prepareStatement(query);
-            preparedStatement.setInt(1, songId);
-            preparedStatement.setInt(2, topic.getId());
+            preparedStatement.setInt(1, song.getId());
+            preparedStatement.setInt(2, song.getTopic().getId());
             preparedStatement.executeUpdate();
             preparedStatement.close();
             return true;
-        }
-        catch (Exception e) {
-            e.printStackTrace();
+        } catch (Exception e) {
+            System.out.println("Exception:" + e);
             return false;
         }
     }
 
-    public int insertTopic(Connection connection, String topic)
+    public Topic insertTopic(Topic topic)
     {
         try {
             String query = "insert into topics (name) values (?)";
             PreparedStatement preparedStatement = connection.prepareStatement(query);
-            preparedStatement.setString(1, topic);
+            preparedStatement.setString(1, topic.getTopic());
             preparedStatement.executeUpdate();
             preparedStatement.close();
+        } catch (Exception e) {
```

Daos shouldn't catch exception. There is no point returning a true or false here. Instead it would be better to just throw the exceptions.